### PR TITLE
qt: add options to enable several modules at once

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **qt/***

categories of modules which can be enabled: essential, addon, deprecated and preview

also, remove a dangling patch

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8343807334/attempts/1)

**Review by jwillikers — APPROVED** [review]

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8343807334/attempts/3)

**Review by valgur — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

File: recipes/qt/5.x.x/conanfile.py [hunk 1]
```diff
@@ -246,6 +241,7 @@ def configure(self):
             modulename = section[section.find('"') + 1: section.rfind('"')]
             status = str(config.get(section, "status"))
             if status not in ("obsolete", "ignore"):
+                assert status in self._module_statuses, f"module {modulename} has status {status} which is not in self._module_statuses {self._module_statuses}"
```
Comment (uilianries) [inline recipes/qt/5.x.x/conanfile.py:244]: Prefer raising `ConanException` here, so we can track it better.

File: recipes/qt/5.x.x/conanfile.py [hunk 2]
```diff
@@ -265,13 +261,27 @@ def _enablemodule(mod):
                 _enablemodule(req)
 
         for module in self._submodules:
+            for status in self._module_statuses:
+                if getattr(self.options, f"{status}_modules"):
+                    if module in submodules_tree and submodules_tree[module]['status'] == status:
+                        _enablemodule(module)
             if self.options.get_safe(module):
```
Comment (uilianries) [inline recipes/qt/5.x.x/conanfile.py:268]:
```suggestion
            module_enabled = False
            for status in self._module_statuses:
                if getattr(self.options, f"{status}_modules"):
                    if module in submodules_tree and submodules_tree[module]['status'] == status:
                        _enablemodule(module)
                        module_enabled = True
                        break
            if self.options.get_safe(module) and not module_enabled:
```
Avoid enabling twice

**Review by uilianries — COMMENTED** [review]
It's a good custom option instead of enabling tons of modules separated. Only few suggestions, but looking good.

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/8569469048/attempts/1)

**Review by jwillikers — APPROVED** [review]

**Pull request merged**